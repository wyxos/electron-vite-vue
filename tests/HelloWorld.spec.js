import { mount } from '@vue/test-utils'
import Hello from '../src/renderer/src/components/HelloWorld.vue'
import { it, expect, describe } from 'vitest'

describe('Component test', () => {
  it('mount component', async () => {
    expect(Hello).toBeTruthy()

    const wrapper = mount(Hello, {
      props: {
        msg: 'This is a test message'
      }
    })

    expect(wrapper.text()).toContain('This is a test message')
    expect(wrapper.html()).toMatchSnapshot()
  }, 0)
})
