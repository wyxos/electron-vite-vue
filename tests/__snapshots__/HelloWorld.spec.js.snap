// Vitest Snapshot v1

exports[`Component test > mount component 1`] = `
"<h1 data-v-24eec2fb=\\"\\">This is a test message</h1>
<p data-v-24eec2fb=\\"\\"> Recommended IDE setup: <a href=\\"https://code.visualstudio.com/\\" target=\\"_blank\\" data-v-24eec2fb=\\"\\">VSCode</a> + <a href=\\"https://github.com/johnsoncodehk/volar\\" target=\\"_blank\\" data-v-24eec2fb=\\"\\">Volar</a></p>
<p data-v-24eec2fb=\\"\\"><a href=\\"https://vitejs.dev/guide/features.html\\" target=\\"_blank\\" data-v-24eec2fb=\\"\\"> Vite Documentation </a> | <a href=\\"https://v3.vuejs.org/\\" target=\\"_blank\\" data-v-24eec2fb=\\"\\">Vue 3 Documentation</a></p><button type=\\"button\\" data-v-24eec2fb=\\"\\">count is: 0</button>
<p data-v-24eec2fb=\\"\\"> Edit <code data-v-24eec2fb=\\"\\">components/HelloWorld.vue</code> to test hot module replacement. </p>"
`;
